### DISCUSSION TITLE
Upscaling of blurry text images

### DISCUSSION THREAD

**Discussion by Brog33** [upscayl/upscayl#686] [Q&A]
i've found that real-esrgan does the better job on blurry images of text than the other models included. but i can't help thinking there must exist a better alternative out there that is especially trained on restoring text/letters. anyone know of such an upscaler, or technique to get better results?

**Comment by NayamAmarshe** [comment]
It'll probably take another model to be trained on such data. I don't know of any such models but I'm sure it can be made.

**Comment by aaronliu0130** [comment]
https://openmodeldb.info/?t=text

**Reply by Brog33** [reply]
wow that is very interesting. thanks for the link.
does upscayl support use of external models?

**Reply by aaronliu0130** [reply]
Yes, there’s a conversion guide in the wiki.

**Reply by Brog33** [reply]
thanks, i followed the guide but the converted model didn't process/output anything.
the only thing i was a bit unsure about in the guide was the text editing bit in the .param file

here is the start of the param file that was generated.

```
7767517
1001 1784
Input            data                     0 1 data
Convolution      /Conv                    1 1 data /Conv_output_0 0=64 1=3 4=1 5=1 6=1728
```

it doesn't match that of in the guide because there are only 'data' statements here after the first 'Input'. i tried to rename the first data to input.1 and also to leave it as is, but it didn't make any difference.

do you have any idea?

**Reply by Brog33** [reply]
actually when looking at the generated .param file, it looks a bit different overall than the one in the guide. could the format have changed since the guide was written?

here are the next few lines:
```
Split            splitncnn_0              1 8 /Conv_output_0 /Conv_output_0_splitncnn_0 /Conv_output_0_splitncnn_1 /Conv_output_0_splitncnn_2 /Conv_output_0_splitncnn_3 /Conv_output_0_splitncnn_4 /Conv_output_0_splitncnn_5 /Conv_output_0_splitncnn_6 /Conv_output_0_splitncnn_7
Convolution      /Conv_1                  1 1 /Conv_output_0_splitncnn_7 /LeakyRelu_output_0 0=32 1=3 4=1 5=1 6=18432 9=2 -23310=1,2.000000e-01
Split            splitncnn_1              1 4 /LeakyRelu_output_0 /LeakyRelu_output_0_splitncnn_0 /LeakyRelu_output_0_splitncnn_1 /LeakyRelu_output_0_splitncnn_2 /LeakyRelu_output_0_splitncnn_3
Concat           /Concat                  2 1 /Conv_output_0_splitncnn_6 /LeakyRelu_output_0_splitncnn_3 /Concat_output_0
Convolution      /Conv_2                  1 1 /Concat_output_0 /LeakyRelu_1_output_0 0=32 1=3 4=1 5=1 6=27648 9=2 -23310=1,2.000000e-01
Split            splitncnn_2              1 3 /LeakyRelu_1_output_0 /LeakyRelu_1_output_0_splitncnn_0 /LeakyRelu_1_output_0_splitncnn_1 /LeakyRelu_1_output_0_splitncnn_2
Concat           /Concat_1                3 1 /Conv_output_0_splitncnn_5 /LeakyRelu_output_0_splitncnn_2 /LeakyRelu_1_output_0_splitncnn_2 /Concat_1_output_0
Convolution      /Conv_3                  1 1 /Concat_1_output_0 /LeakyRelu_2_output_0 0=32 1=3 4=1 5=1 6=36864 9=2 -23310=1,2.000000e-01
```

vs

![image](https://github.com/upscayl/upscayl/assets/1303537/618630f0-a4f6-4bed-a041-f2e87f9309e7)

**Reply by aaronliu0130** [reply]
@NayamAmarshe Could you look into this and #690?

**Reply by NayamAmarshe** [reply]
The model needs to be for ESRGAN specifically

**Reply by Brog33** [reply]
it says ESRGAN. this is the one i try to convert: https://openmodeldb.info/models/8x-NMKD-Typescale

**Reply by NayamAmarshe** [reply]
> it says ESRGAN. this is the one i try to convert: https://openmodeldb.info/models/8x-NMKD-Typescale
> 

8x models aren't supported yet.

**Reply by Brog33** [reply]
okay. however the same issue occurs for the 2x model i converted too. https://openmodeldb.info/models/2x-BSTexty
i suspect the generated .param file is the culprit.

**Reply by NayamAmarshe** [reply]
Make sure that 2x, 1x or 3x models have the scale in the file names.
For example, the name of the converted model should be something like: bstexty-2x.

**Reply by Brog33** [reply]
the filename was 2x_BSTexty.bin / 2x_BSTexty.param
i did a new conversion with the name bstexty-2x, but it made no difference

![image](https://github.com/upscayl/upscayl/assets/1303537/d8119f64-4ebf-4ad9-84f7-033640884545)

**Reply by NayamAmarshe** [reply]
Can you zip it and send the bin and param?

**Reply by Brog33** [reply]
certainly. these are the raw files after the chainner conversion (i have not edited the params file)
https://wormhole.app/DNo0R#gPap6mmV4B-J9MKYRn1yVQ

**Reply by Brog33** [reply]
any progress on this?
would be nice with an updated guide on how to convert models.

**Reply by aaronliu0130** [reply]
We've recently updated the guide; chaiNNer versions above and including 21.2 may not work

**Comment by ArshCypherZ** [comment]
yes, personally i like [Enhance AI's upscaling](https://enhanceai.art/upscaling)

**Reply by aaronliu0130** [reply]
Are you sure it upscales text optimally? Didn't really work on my try
